legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java

```diff
@@ -25,16 +25,38 @@ public class GraphQLProdCacheKey implements GraphQLCacheKey
     private String runtimePath;
     private String queryClassPath;
     private String query;
+    private String dataspacePath;
+    private String executionContext;
 
-    public GraphQLProdCacheKey(String groupID, String artifactId, String versionId, String mappingPath, String runtimePath, String queryClassPath, String query)
+    GraphQLProdCacheKey()
     {
-        this.groupID = groupID;
-        this.artifactId = artifactId;
-        this.versionId = versionId;
-        this.mappingPath = mappingPath;
-        this.runtimePath = runtimePath;
-        this.queryClassPath = queryClassPath;
-        this.query = query;
+
+    }
+
+    public static GraphQLProdCacheKey newGraphQLProdCacheKey(String groupID, String artifactId, String versionId, String mappingPath, String runtimePath, String queryClassPath, String query)
+    {
+        GraphQLProdCacheKey graphQLProdCacheKey = new GraphQLProdCacheKey();
+        graphQLProdCacheKey.groupID = groupID;
+        graphQLProdCacheKey.artifactId = artifactId;
+        graphQLProdCacheKey.versionId = versionId;
+        graphQLProdCacheKey.mappingPath = mappingPath;
+        graphQLProdCacheKey.runtimePath = runtimePath;
+        graphQLProdCacheKey.queryClassPath = queryClassPath;
+        graphQLProdCacheKey.query = query;
+        return graphQLProdCacheKey;
+    }
+
+    public static GraphQLProdCacheKey newGraphQLProdCacheKeyWithDataspace(String groupID, String artifactId, String versionId, String dataspacePath, String executionContext, String queryClassPath, String query)
+    {
+        GraphQLProdCacheKey graphQLProdCacheKey = new GraphQLProdCacheKey();
+        graphQLProdCacheKey.groupID = groupID;
+        graphQLProdCacheKey.artifactId = artifactId;
+        graphQLProdCacheKey.versionId = versionId;
+        graphQLProdCacheKey.dataspacePath = dataspacePath;
+        graphQLProdCacheKey.executionContext = executionContext;
+        graphQLProdCacheKey.queryClassPath = queryClassPath;
+        graphQLProdCacheKey.query = query;
+        return graphQLProdCacheKey;
```

Comment:
Would rather prefer to have two sub classes of cache key - one for mapping, runtime and one for query, unless there is an advantage of this approach

Reply:
Done